src/events/kafkajs.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Handlers for kafkajs instrumentation
+ */
+
+const uuid4 = require('uuid4');
+const tracer = require('../tracer.js');
+const eventInterface = require('../event.js');
+const moduleUtils = require('./module_utils.js');
+const { EPSAGON_HEADER } = require('../consts.js');
+
+
+/**
+ * acts as a middleware for `producer.send()`
+ * @param {object} messages the messages param to send
+ * @param {Kafka} producer producer object
+ * @returns {Promise} The response promise
+ */
+function kafkaMiddleware(messages, producer) {
+    let response = Promise.resolve();
+    try {
+        const { slsEvent: kafkaEvent, startTime } = eventInterface.initializeEvent(
+            'kafka',
+            messages.topic,
+            'send',
+            'kafkajs'
+        );
+
+        const epsagonId = uuid4();
+        // eslint-disable-next-line no-param-reassign
+        messages.messages = messages.messages.map((message) => {
+            if (!message.headers) {
+                // eslint-disable-next-line no-param-reassign
+                message.headers = {};
+            }
+            // eslint-disable-next-line no-param-reassign
+            message.headers[EPSAGON_HEADER] = epsagonId;
+            return message;
+        });
+
+        response = producer.originalSend(messages);
+
+        const sendPromise = new Promise((resolve) => {
+            let result;
+            let originalHandlerAsyncError;
+            response.then((res) => {
```

Comment:
you are doing promise parallelizing here (which is not appropriate for this case) either call `originalSend` here or override response

Reply:
👍